### ISSUE TITLE
Member selection dialog for Pull Members Up, Extract Base Class, and Move Static Members should warn if dependents aren't selected

### ISSUE THREAD

**Issue by Soreloser2** [dotnet/roslyn#56226]
**Problem Statement:**
Currently, when we refactor members through Move Static Members, Pull Members Up, or Extract Base Class using the dialog option, there is an option to select dependents. This means any applicable members that were referenced by our current selection get added to the selection. For example:
```
class C
{
    public const int NUM = 0;

    public static int M()
    {
        return NUM;
    }
}
```
If we apply "extract base class" to `M()` and select only `M()` as our member, then when we press "Select Dependents", `NUM` also is added to our selection. This ensures that when we move `M()` to a base class, the reference of `NUM` will still be qualified. 

However, if the user did not press "Select Dependents", `NUM` would not be moved, meaning the reference in `M()` would give an error: "The name "NUM" does not exist in the current context". It may be confusing to a user as to why `NUM` wasn't requalified so that the reference worked. 

We don't refactor for a number of reasons, mainly because it's not a good practice (and sometimes impossible) to have a base class depend upon a specific derived class.

**Suggestion/Request:**
The suggestion is to add some sort of warning icon to the member selection dialog to indicate when a selection's dependent is unselected. It would only appear when the member that has dependents is not being made abstract (i.e. the body of the method, which contains the references, is actually being moved).

These warnings wouldn't prevent the action from taking place, and so would probably be yellow/informational warnings. There are good reasons why somebody wouldn't want to select a dependent like that, but making them aware that the action could produce an error, so that they are prepared to deal with it, would be useful.

**UX/Implementation Details:**
 The icon could go near the glyph for the member type, and have a tooltip that reads something to the effect of "This member is used by {using member or list of using members} and may be unqualified after movement." 

Another option is to have the warning on the selected member itself, saying something like "This member depends on other members and may have unqualified references after movement, consider selecting dependents" 

Both of these messages are probably too wordy to fit in a tooltip, but maybe a shortened version of either of those, or the text could be displayed below the selection box as a separate message with the same icon.

Pull Members Up and Extract Base Class both use the same selection dialog component, while Move Static Members uses a different one. Part of the addition could include merging the Move Static Members selection component with the other two, or it could simply be adding the same warning/icon to both components.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback about this issues.
